aiocache/backends/memcached.py

```diff
@@ -139,16 +142,22 @@ class MemcachedCache(MemcachedBackend):
     """
 
     NAME = "memcached"
+    KeyType = Union[bytes, bytes]  # Workaround: TypeAlias not in python <= 3.10
 
+    # def __init__(self, serializer=None, *, key_type: CacheKeyType = bytes, **kwargs):
     def __init__(self, serializer=None, **kwargs):
         super().__init__(serializer=serializer or JsonSerializer(), **kwargs)
 
     @classmethod
     def parse_uri_path(cls, path):
         return {}
 
-    def _build_key(self, key, namespace=None):
-        ns_key = super()._build_key(key, namespace=namespace).replace(" ", "_")
+    # def build_key(self, key: str, namespace: Optional[str] = None) -> CacheKeyType:
+    #     ns_key = super().build_key(key, namespace=namespace).replace(" ", "_")
+    #     return str.encode(ns_key)
+
+    def build_key(self, key: str, namespace: Optional[str] = None) -> KeyType:
+        ns_key = cast(str, super().build_key(key, namespace=namespace)).replace(" ", "_")
```

Comment:
An alternative to `cast` could simply be `# type: ignore`

Reply:
Hmm, yes, that's slightly awkward.

Maybe a type safe option would be to make the `BaseCache` define it as an abstract method, and provide this default method under another name (like _str_build_key() or something)?

Reply:
> Maybe a type safe option would be to make the `BaseCache` define it as an abstract method, and provide this default method under another name (like `_str_build_key()` or something)?

I think you're right.  I've been trying to make this work by overloading `build_key()` with `@singledispatchmethod` to no avail.  For that I would need to implement some factory function that returns a `BaseCache` with the appropriate return type for `build_key()`.  Implementing this private helper method seems much simpler.

Reply:
`build_key()` is now an abstract method on `BaseCache`; and overloaded/defined for the derived classes.